v2/errors.go

```diff
@@ -53,9 +53,9 @@ func (e HTTPStatusCodeError) Error() string {
 // IsHTTPError returns whether the error represents an HTTPStatusCodeError.  A
 // client method returning an HTTP error indicates that the broker returned an
 // error code and a correctly formed response body.
-func IsHTTPError(err error) bool {
-	_, ok := err.(HTTPStatusCodeError)
-	return ok
+func IsHTTPError(err error) (HTTPStatusCodeError, bool) {
```

Comment:
Let's return a pointer here so we can return null if the conversion isn't valid.  I guess that means you will have to check conversions to both `HTTPStatusCodeError` and `*HTTPStatusCodeError`